### PR TITLE
Add commandui class

### PR DESCRIPTION
**Pull request by evan-m-jackson**

### CODE REVIEW

File: spec/game_over_spec.rb [hunk 1]
```diff
@@ -1,12 +1,26 @@
 require 'rspec'
 require './lib/board'
 require './lib/game_over'
+require './spec/player_choice_spec'
+
+class TestUI
```
Comment (jchung722) [inline spec/game_over_spec.rb:6]: nice!

**Review by jchung722 — APPROVED** [review]
👏

File: lib/player_choice.rb [hunk 2]
```diff
@@ -28,9 +29,9 @@ def is_space_free(board, choice)
 
   def move_confirmation_message(board, choice)
     if !is_user_input_valid(choice)
-      puts "Sorry that is not a valid move. Please select an integer between 1 and 9.\n"
+      @output.sorry_invalid
     elsif !is_space_free(board, choice)
-      puts 'Sorry this space is taken. Please select another.'
+      @output.sorry_taken
```
Comment (skp96) [inline lib/player_choice.rb:34]: Excellent job moving output-related functionality into its own class and creating well-defined method names!

File: spec/player_choice_spec.rb [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
+  def sorry_invalid
+    @sorry_invalid_move = true
+  end
+end
 
 describe 'PlayerChoice' do
   before(:each) do
-    @choice = PlayerChoice.new
+    @test_ui = TestUI.new
+    @choice = PlayerChoice.new(output: @test_ui)
```
Comment (skp96) [inline spec/player_choice_spec.rb:21]: Great job injecting the `TestUI` class in order to test output without cluttering test runs with `put` statements!

**Review by skp96 — APPROVED** [review]
Awesome work! Refactoring is coming along nicely and your code is becoming more readable than before. By the way, feel free to use the PR description area to leave notes on changes you're proud of that you'd want to highlight, or understanding of principles/knowledge you've applied, and how. Just to note, it's not an ask. It's really for you.

**Pull request merged**